janus.c

```diff
@@ -191,16 +191,42 @@ static gchar *local_ip = NULL;
 gchar *janus_get_local_ip(void) {
 	return local_ip;
 }
-static gchar *public_ip = NULL;
+static GPtrArray *public_ip = NULL;
+guint janus_get_public_ip_count(void) {
+	return public_ip ? public_ip->len : 0;
+}
+gchar *janus_get_public_ip_index(guint index) {
+	if (!public_ip || !public_ip->len) {
+		/* Fallback to the local IP, if we have no public one */
+		return local_ip;
+	}
+	if (index >= public_ip->len) {
+		return NULL;
+	}
+	return g_ptr_array_index(public_ip, index);
+}
 gchar *janus_get_public_ip(void) {
-	/* Fallback to the local IP, if we have no public one */
-	return public_ip ? public_ip : local_ip;
+	return janus_get_public_ip_index(0);
+}
```

Comment:
I see you're keeping the old method as a fallback, but since it's internal stuff, it's not like we really need to keep a strong backwards compatibility. I'd say let's keep it simple and just have a single method there, which references indexes.